File: 2nd year/Laboratories/MAP/L9/CRUD/Domain/Validator/MarkValidator.cs
using static CRUD.Utils.WeekUtils;

namespace CRUD.Domain.Validator
{
    public class MarkValidator : IValidator<Mark, string>
    {
        public void Validate(Mark mark)
        {
            if (mark.Id == null || mark.TaskId < 1 || mark.StudentId < 1)
            {
                throw new ValidationException("Id is invalid (null or negative)");
            }

            if (mark.Value < 1 || mark.Value > 10)
            {
                throw new ValidationException("Value is invalid (not between 1 - 10)");
            }

            if (mark.Week < 1 || mark.Week > SemesterWeekSpan)
            {
                throw new ValidationException($"Week is invalid (not between 1 - {SemesterWeekSpan})");
            }

            if (string.IsNullOrEmpty(mark.Feedback))
            {
                throw new ValidationException("Feedback is invalid (null or empty)");
            }
        }
    }
}

File: 2nd year/Laboratories/MAP/7/src/main/java/domain/validator/MarkValidator.java
package domain.validator;

import domain.Mark;

public class MarkValidator implements Validator<Mark> {
    @Override
    public void validate(Mark mark) throws ValidationException {
        // validate id
        String id = mark.getId();
        Integer taskId = mark.getTaskId();
        Integer studentId = mark.getStudentId();
        if (id == null || taskId < 0 || studentId < 0) {
            throw new ValidationException("Id is invalid (null or negative)");
        }

        // validate value
        Double value = mark.getValue();
        if (value == null || value < 1 || value > 10) {
            throw new ValidationException("Value is invalid (null or not between 1-10)");
        }

        // validate week
        Integer week = mark.getWeek();
        if (week == null || week < 1 || week > 28) {
            throw new ValidationException("Week is invalid (null or not between 1-28)");
        }

        // validate feedback
        String feedback = mark.getFeedback();
        if (feedback == null || feedback.length() == 0) {
            throw new ValidationException("Feedback is invalid (null or empty)");
        }
    }
}
